src/tools/live-utils/start-session.ts

```diff
@@ -1,47 +1,106 @@
 import { sanitizeUrlParam } from "../../lib/utils";
 import logger from "../../logger";
 import childProcess from "child_process";
+import { filterDesktop } from "./desktop-filter";
+import { filterMobile } from "./mobile-filter";
+import {
+  DesktopArgs,
+  MobileArgs,
+  DesktopEntry,
+  MobileEntry,
+  PlatformType,
+} from "./types";
+import {
+  isLocalURL,
+  ensureLocalBinarySetup,
+  killExistingBrowserStackLocalProcesses,
+} from "../../lib/local";
 
-interface StartSessionArgs {
-  browser: string;
-  os: string;
-  osVersion: string;
-  url: string;
-  browserVersion: string;
-  isLocal: boolean;
+/**
+ * Prepares local tunnel setup based on URL type
+ */
+async function prepareLocalTunnel(url: string): Promise<boolean> {
+  const isLocal = isLocalURL(url);
+  if (isLocal) {
+    await ensureLocalBinarySetup();
+  } else {
+    await killExistingBrowserStackLocalProcesses();
+  }
+  return isLocal;
 }
 
+/**
+ * Entrypoint: detects platformType & delegates.
+ */
 export async function startBrowserSession(
-  args: StartSessionArgs,
+  args: DesktopArgs | MobileArgs,
 ): Promise<string> {
-  // Sanitize all input parameters
-  const sanitizedArgs = {
-    browser: sanitizeUrlParam(args.browser),
-    os: sanitizeUrlParam(args.os),
-    osVersion: sanitizeUrlParam(args.osVersion),
+  const entry =
+    args.platformType === PlatformType.DESKTOP
+      ? await filterDesktop(args as DesktopArgs)
+      : await filterMobile(args as MobileArgs);
+
+  const isLocal = await prepareLocalTunnel(args.url);
+
+  const url =
+    args.platformType === PlatformType.DESKTOP
+      ? buildDesktopUrl(args as DesktopArgs, entry as DesktopEntry, isLocal)
+      : buildMobileUrl(args as MobileArgs, entry as MobileEntry, isLocal);
+
+  openBrowser(url);
+  return entry.notes ? `${url}, ${entry.notes}` : url;
+}
+
+function buildDesktopUrl(
+  args: DesktopArgs,
+  e: DesktopEntry,
+  isLocal: boolean,
+): string {
+  const params = new URLSearchParams({
+    os: sanitizeUrlParam(e.os),
```

Comment:
With `URLSearchParams`, `sanitizeUrlParam` is no longer required I guess, confirm once